### ISSUE TITLE
Grid multiple selection does not work correctly anymore in version 7.2.0

### ISSUE THREAD

**Issue by plevamatus** [telerik/kendo-react#2006]
## I'm submitting a...

* Regression report 

## Current behavior
When I click the "Select All" checkbox in the header, all rows in the grid get highlighted, but the checkboxes themselves remain unchecked.
<img width="380" alt="image" src="https://github.com/telerik/kendo-react/assets/11454499/28ac2279-543a-4039-b969-6bef8e13fdad"><img width="365" alt="image" src="https://github.com/telerik/kendo-react/assets/11454499/196ae477-1d6f-419a-a8e6-d2a4b842cb23">

Similarly, when I click on a row, it gets highlighted, but the checkbox remains unchecked.
<img width="365" alt="image" src="https://github.com/telerik/kendo-react/assets/11454499/28ab5ce2-41f9-433c-971b-bb4e01c30d6e">

## Expected behavior
I would expect all highlighted rows to have their checkboxes automatically checked.
<img width="364" alt="image" src="https://github.com/telerik/kendo-react/assets/11454499/6869a15a-33dc-4658-b12b-4a046d5719d3">

## Minimal reproduction of the problem with instructions
Open this example (https://www.telerik.com/kendo-react-ui/components/grid/selection/#toc-customizing-the-selection) from your documentation and try clicking on the "Select All" checkbox, or on any row.
https://codesandbox.io/p/sandbox/mystifying-sun-9ym7zl

## Environment

Package versions:

```
+-- @progress/kendo-data-query@1.7.0
+-- @progress/kendo-date-math@1.5.12
+-- @progress/kendo-drawing@1.19.0
+-- @progress/kendo-inputs-common@3.1.0
+-- @progress/kendo-intl@3.1.2
+-- @progress/kendo-licensing@1.3.5
+-- @progress/kendo-popup-common@1.9.2
+-- @progress/kendo-react-animation@7.2.0
+-- @progress/kendo-react-buttons@7.2.0
+-- @progress/kendo-react-common@7.2.0
+-- @progress/kendo-react-data-tools@7.2.0
+-- @progress/kendo-react-dateinputs@7.2.0
+-- @progress/kendo-react-dialogs@7.2.0
+-- @progress/kendo-react-dropdowns@7.2.0
+-- @progress/kendo-react-form@7.2.0
+-- @progress/kendo-react-grid@7.2.0
+-- @progress/kendo-react-inputs@7.2.0
+-- @progress/kendo-react-intl@7.2.0
+-- @progress/kendo-react-labels@7.2.0
+-- @progress/kendo-react-layout@7.2.0
+-- @progress/kendo-react-popup@7.2.0
+-- @progress/kendo-react-progressbars@7.2.0
+-- @progress/kendo-react-treeview@7.2.0
+-- @progress/kendo-svg-icons@2.1.0
+-- prop-types@15.8.1
+-- react@18.2.0
+-- react-dom@18.2.0
+-- react-transition-group@2.9.0
```

Browser:

- Chrome (desktop) version121.0.6167.185

System:
- Platform:  Windows

**Comment by iadnonov** [comment]
Hey, @plevamatus

Thank you for bringing this behavior to our attention. This is a known issue which was fixed in v7.1.0, but it seems that appears again with version v7.2.0. 

As a temporary workaround, you can switch back to v7.1.0 https://stackblitz.com/edit/react-p2gquf-qyxmts?file=app%2Fmain.jsx until our team fix the issue.

Please excuse us for any inconvenience caused by this and if you have any other questions, please don't hesitate to submit a ticket in [our support system](https://www.telerik.com/account/support-center) where we do our best to answer all possible questions regarding KendoReact suite.

Greetings,
Ivaylo

**Comment by iadnonov** [comment]
Fixed in dev version 7.2.1-develop.1

**Comment by elena-gancheva** [comment]
The issue is fixed in the latest official version of the package: [`7.2.1`](https://github.com/telerik/kendo-react-private/releases/tag/v7.2.1)
